### PR TITLE
Feat: check service availability

### PR DESCRIPTION
**Pull request by Gamboster**

### CODE REVIEW

**Review by gabrielbazan7 — DISMISSED** [review]
ACK!

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 1]
```diff
@@ -5029,6 +5029,32 @@ export class WalletService {
     });
   }
 
+  checkServiceAvailability(req): Promise<boolean> {
+    return new Promise((resolve, reject) => {
```
Comment (kajoseph) [inline packages/bitcore-wallet-service/src/lib/server.ts:5033]: I'd like to move away from this sort of verbosity. Could you instead just make this function an `async` function and throw/return below?

File: packages/bitcore-wallet-service/src/lib/expressapp.ts [hunk 2]
```diff
@@ -1315,6 +1315,23 @@ export class ExpressApp {
       });
     });
 
+    router.post('/v1/service/checkAvailability', (req, res) => {
```
Comment (kajoseph) [inline packages/bitcore-wallet-service/src/lib/expressapp.ts:1318]: I'd like to move away from promise chains and callbacks and instead use async/await syntax. Could you make this router function `async` and extend the try..catch below to encompass the whole function and `await server.checkServiceAvailability`?

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 3]
```diff
@@ -5029,6 +5029,32 @@ export class WalletService {
...
+      if (!checkRequired(req.body, ['service', 'opts'])) {
+        return reject(new ClientError('checkServiceAvailability request missing arguments'));
+      }
+
+      let serviceEnabled: boolean;
+
+      switch (req.body.service) {
+        case '1inch':
+          if (req.body.opts && req.body.opts.country && req.body.opts.country.toUpperCase() == 'US') {
```
Comment (kajoseph) [inline packages/bitcore-wallet-service/src/lib/server.ts:5042]: This can be shortened to `req.body.opts?.country?.toUpperCase() === 'US'`

**Review by kajoseph — CHANGES_REQUESTED** [review]

File: packages/bitcore-wallet-service/src/lib/server.ts (hunk 3 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/lib/server.ts]: Done. Thanks

File: packages/bitcore-wallet-service/src/lib/expressapp.ts (hunk 2 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/lib/expressapp.ts:1318]: Fixed. I avoided using promises for this particular function

File: packages/bitcore-wallet-service/src/lib/server.ts (hunk 1 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/lib/server.ts]: Fixed. I avoided using promises for this particular function

**Review by kajoseph — APPROVED** [review]
Looks good! Thanks

**Pull request merged**